src/compas_fab/robots/targets.py

```diff
@@ -0,0 +1,449 @@
+                raise NotImplementedError("Unsupported joint type: {}".format(joint_type))
+        return tolerances_above, tolerances_below
+
+    def scaled(self, factor):
+        """Returns copy of the target where the target configuration and tolerances are scaled.
+
+        This function should only be needed if the ConfigurationTarget was created
+        with a distance unit other than meters.
+
+        Only the values for prismatic and planar joints are scaled. The values for revolute
+        and continuous joints are not scaled, as they must be in radians.
+
+        Parameters
+        ----------
+        factor : float
+            The scaling factor.
+
+        Returns
+        -------
+        :class:`ConfigurationTarget`
+            The scaled configuration target.
+        """
+        target_configuration = self.target_configuration.scaled(factor)
+
+        def scale_tolerance(tolerance, joint_types):
+            scaled_tolerance = []
+            for t, joint_type in zip(tolerance, joint_types):
+                if joint_type in (Joint.PLANAR, Joint.PRISMATIC):
+                    t *= factor
+                scaled_tolerance.append(t)
+            return scaled_tolerance
+
+        # We scale only the tolerances for prismatic and planar joints,
+        # similar to the Configuration.scale() method
+        tolerance_above = (
+            scale_tolerance(self.tolerance_above, target_configuration.joint_types) if self.tolerance_above else None
+        )
+        tolerance_below = (
+            scale_tolerance(self.tolerance_below, target_configuration.joint_types) if self.tolerance_below else None
+        )
+
+        return ConfigurationTarget(target_configuration, tolerance_above, tolerance_below, self.name)
+
+
+class ConstraintSetTarget(Target):
+    """Represents a list of Constraint as the target for motion planning.
+
+    Given a ConstraintSetTarget, the planner aims to find a path moving
+    the robot's end-effector to satisfy ALL the constraints in the constraint set.
+    Constraints can be very specific, for example defining value domains
+    for each joint, such that the goal configuration is included,
+    or defining a volume in space, to which a specific robot link (e.g.
+    the end-effector) is required to move to.
+
+    This target cannot be translated into a single pose or configuration target
+    except in trivial cases. Therefore the ending pose or configuration of the
+    planned path can only be determined after performing the motion planning.
+
+    ConstraintSetTarget is suitable for advanced users who want to specify
+    custom constraints for the robot motion planning.
+    Different planner backends may support differnt types of Constraints.
+    See tutorial :ref:`targets` for more details.
+
+    ConstraintSetTarget is only supported by Free motion planning (using the MoveIt Planner),
+    Inverse Kinematics and Cartesian motion planning do not support this target type.
```

Comment:
I wouldn't mention one specific planner here. Also IK is never target-related, it's always `Frame` to `Configuration`, so, I would remove it from the sentence:

```suggestion
    ConstraintSetTarget is only supported by Free motion planning,
    Cartesian motion planning do not support this target type.
```